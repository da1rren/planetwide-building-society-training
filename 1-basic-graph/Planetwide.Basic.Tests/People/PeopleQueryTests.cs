
using Microsoft.AspNetCore.Mvc.Testing;

namespace Planetwide.Basic.Tests.People;

public class PeopleQueryTests : IClassFixture<WebServerFixture>
{
    internal WebApplicationFactory<Program> Server { get; init; }

    public HttpClient Client { get; init; }

    public PeopleQueryTests(WebServerFixture fixture)
    {
        Server = fixture.Server;
        Client = fixture.Client;
    }

    [Fact]
    public async Task Ensure_Get_Person_Returns_Person()
    {
        // The web app factory should be extracted to a fixture 

        var request = await Client.PostAsJsonAsync("/graphql", new
        {
            query = @"{ people{ id name } }"
        });

        var response = await request.Content.ReadAsStringAsync();
        request.EnsureSuccessStatusCode();
        Snapshot.Match(response);
    }
}

