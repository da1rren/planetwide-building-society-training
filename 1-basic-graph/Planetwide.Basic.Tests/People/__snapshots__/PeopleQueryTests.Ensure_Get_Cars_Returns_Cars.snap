{"data":{"cars":[{"model":"Volvo","maxKph":180},{"model":"Bmw","maxRange":40}]}}
